backend/latest/src/plugin.ts

```diff
@@ -75,91 +71,105 @@ const plugins: BackendPlugins = {
       related_record_id: { equal_any: lines.map(({ id }) => id) },
     });
 
-    // Sqlite only
-    const sql_statement = `
+    const sql_result = getAggregatedAmc(
+      requisition.store_id,
+      lines.map(({ item_link_id }) => item_link_id)
+    );
+
+    return {
+      transformed_lines: lines.map(line => {
+        const max_quantity =
+          line.average_monthly_consumption *
+          (requisition.max_months_of_stock + months_lead_time);
+        const difference = max_quantity - line.available_stock_on_hand;
+        const suggested_quantity = difference < 0 ? 0 : difference;
+        return { ...line, suggested_quantity };
+      }),
+      // Here we need to do sql query for plugin data (another reason why plugin data should be on the record)
+      plugin_data: lines.map(line => ({
+        id:
+          pluginData.find(
+            ({ related_record_id }) => related_record_id === line.id
+          )?.id || uuidv7(),
+        store_id: requisition.store_id,
+        plugin_code: pluginCode,
+        related_record_id: line.id,
+        // need to share this
+        data_identifier: dataIdentifier,
+        data: String(
+          sql_result
+            .filter(({ item_id }) => item_id === line.item_link_id)
+            .reduce((acc, row) => acc + row.average_monthly_consumption, 0)
+        ),
+      })),
+    };
+  },
+  graphql_query: ({ store_id, input: inputUntyped }): Graphql['output'] => {
+    const input = inputUntyped as Graphql['input'];
+
+    switch (input.type) {
+      case 'echo':
+        return { type: 'echo', echo: input.echo };
+      case 'aggregateAmc':
+        return {
+          type: 'aggregateAmc',
+          stats: getAggregatedAmc(store_id, input.itemIds).map(row => ({
+            itemId: row.item_id,
+            amc: row.average_monthly_consumption,
+            name: row.name,
+          })),
+        };
+      default:
+        assertUnreachable(input);
+    }
+  },
+};
+
+const getAggregatedAmc = (storeId: string, itemIds: string[]) => {
+  // Sqlite only
+  return sqlQuery(
+    ['item_id', 'average_monthly_consumption', 'name'],
+    `
       WITH 
       rl_all AS (
       SELECT 
         rl.average_monthly_consumption,
         il.item_id,
         r.store_id,
         nl.name_id,
+        n.name,
```

Comment:
The sql statement was relocated and changed a bit, no 'summed', not worried to much about item links and return name. Why not worrying too much about item links ? (we should be, I really need to make an issue about simplifying plugin examples, civ has a good example of how and when item links should be used in sql queries)

Reply:
Hehe yeah would be good to simplify a bit, show the contracts with complex logic 🙏